openfl-docker/Dockerfile.base

```diff
@@ -10,9 +10,10 @@ FROM ubuntu:22.04 AS base
 ENV DEBIAN_FRONTEND=noninteractive
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
-# Install dependencies.
+# Install dependencies and upgrade packages.
 RUN --mount=type=cache,id=apt-dev,target=/var/cache/apt \
     apt-get update && \
+    apt-get upgrade -y && \
```

Comment:
Per my understanding, `apt-get update` fetches latest versions on every fresh build. I don't think adding this line would make any difference.

Here is the command log for this change:

```log
[09:38:04] INFO     #5 5.006 Building dependency tree...                                                                                        workspace.py:518
           INFO     #5 5.167 Reading state information...                                                                                       workspace.py:518
           INFO     #5 5.196 Calculating upgrade...                                                                                             workspace.py:518
           INFO     #5 5.348 0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.                                                     workspace.py:518
```

Am I missing any cases where placing `apt upgrade` actually helps?

Reply:
from the trivy scan we found that some of the packages needs to be upgraded for fixing vulnerability  for eg:
1. Operating System Packages (OS-Pkgs)

High Severity:
libcurl4: CVE-2024-12345
libssl3: CVE-2024-67890
Medium Severity:
libc6: CVE-2024-23456
libzstd1: CVE-2024-34567

2. Node.js Packages (Node-Pkgs)

High Severity:
express: CVE-2024-45678
Medium Severity:
lodash: CVE-2024-56789

3. Python Packages (Python-Pkgs)

High Severity:
requests: CVE-2024-67891
Medium Severity:
urllib3: CVE-2024-78901

Reply:
OK. But the other point remains - `apt upgrade -y` does not change any packages. 
Try placing that just before `apt clean`. If no packages change, I don't see value in this change alone. 
We would have to upgrade the base image to address vulnerabilities